### ISSUE TITLE
Why the prediction2 output is

### ISSUE THREAD

**Issue by whfei561** [zhanruicai/CPC_code#1]
When I change the distribution, let X and Y follow the m-dependent distribution, but I still get NAN when I run the code. I run the code interactively with R and python.
Below is my code:
R:

library(MASS)
#m is simulate 
m       <-   250
#n is sample size
n          <-   1000
#p is d1
p          <-   100
#q is d2
q          <-   100
#M =3  means 3-dependent 
epsilon <-   rnorm(n*(p+q+2*M),0,1)
  E       <-   matrix(epsilon[1:(n*(p+M))],n,p+M)  
  F      <-   matrix(epsilon[(n*(p+M)+1):(n*(q+p+2*M))],n,q+M) 
  x       <-   matrix(0,n,p)
  y       <-   matrix(0,n,q)
  for (i in 1:p){
    x[,i] <-   apply(E[,(i+1):(i+M)],1,sum)
  }
  for (i in 1:q){
    y[,i] <-   apply(F[,(i+1):(i+M)],1,sum)
  }     
Then python :

def get_data(x, y):
    n1 =x.shape[0]

    x1 = np.concatenate((x[1:n1,], x[0:1,]), axis = 0)
        
    xall = np.concatenate((x, x1), axis = 0)
    yall = np.concatenate((y, y), axis = 0)
    
    datall = np.concatenate((xall, yall), axis = 1)
            
        
    label1 = np.ones(shape = n1)
    label0 = np.zeros(shape = n1)
    label = np.concatenate((label1, label0), axis = 0)
            
    return datall, label

import os
import time
import sys
import numpy as np
from statsmodels.distributions.empirical_distribution import ECDF
from statistics import mean
from scipy.stats import norm
import new_function as dg
import tensorflow as tf

x=r.x
y=r.y
sample_size=int(x.shape[0])
d=int(x.shape[1])
train_data_size = sample_size//2
val_data_size = sample_size//2
batchsize = 3

epochs = 20

bei = 40
SGD_rate = 0.1

train_x, train_ground_truth_class = dg.get_data(x[0:train_data_size,],y[0:train_data_size,])
val_x, val_ground_truth_class = dg.get_data(x[train_data_size:sample_size,], y[train_data_size:sample_size,])
        
model = tf.keras.models.Sequential([
tf.keras.layers.Dense(d*bei, activation='relu', kernel_regularizer=tf.keras.regularizers.L1(0.001)),
# tf.keras.layers.Dense(d*bei, activation='relu'),
tf.keras.layers.Dropout(0.3),
tf.keras.layers.Dense(1, activation='sigmoid')])        
        
# Create an optimizer with the desired parameters.
myopt = tf.keras.optimizers.SGD(learning_rate = SGD_rate)

model.compile(optimizer=myopt, loss='binary_crossentropy', metrics=['accuracy'])
        
model.fit(train_x, train_ground_truth_class,  epochs=epochs, batch_size=batchsize, verbose = 0)

prediction2 = model.predict(val_x)

Then, print(prediction2) are all NAN, why?

**Comment by whfei561** [comment]
When I change the distribution, let X and Y follow the m-dependent distribution, but I still get NAN when I run the code. I run the code interactively with R and python.
Below is my code:
R:

library(MASS)
#m is simulate 
m       <-   250
#n is sample size
n          <-   1000
#p is d1
p          <-   100
#q is d2
q          <-   100
#M =3  means 3-dependent 
epsilon <-   rnorm(n*(p+q+2*M),0,1)
  E       <-   matrix(epsilon[1:(n*(p+M))],n,p+M)  
  F      <-   matrix(epsilon[(n*(p+M)+1):(n*(q+p+2*M))],n,q+M) 
  x       <-   matrix(0,n,p)
  y       <-   matrix(0,n,q)
  for (i in 1:p){
    x[,i] <-   apply(E[,(i+1):(i+M)],1,sum)
  }
  for (i in 1:q){
    y[,i] <-   apply(F[,(i+1):(i+M)],1,sum)
  }     
Then python :

def get_data(x, y):
    n1 =x.shape[0]

    x1 = np.concatenate((x[1:n1,], x[0:1,]), axis = 0)
        
    xall = np.concatenate((x, x1), axis = 0)
    yall = np.concatenate((y, y), axis = 0)
    
    datall = np.concatenate((xall, yall), axis = 1)
            
        
    label1 = np.ones(shape = n1)
    label0 = np.zeros(shape = n1)
    label = np.concatenate((label1, label0), axis = 0)
            
    return datall, label

import os
os.chdir("C:/Users/PC/Desktop/recently/rank independence test with m-dependence/code")

import time
import sys
import numpy as np
from statsmodels.distributions.empirical_distribution import ECDF
from statistics import mean
from scipy.stats import norm
import new_function as dg
import tensorflow as tf

x=r.x
y=r.y
sample_size=int(x.shape[0])
d=int(x.shape[1])
train_data_size = sample_size//2
val_data_size = sample_size//2
batchsize = 3

epochs = 20

bei = 40
SGD_rate = 0.1

train_x, train_ground_truth_class = dg.get_data(x[0:train_data_size,],y[0:train_data_size,])
val_x, val_ground_truth_class = dg.get_data(x[train_data_size:sample_size,], y[train_data_size:sample_size,])
        
model = tf.keras.models.Sequential([
tf.keras.layers.Dense(d*bei, activation='relu', kernel_regularizer=tf.keras.regularizers.L1(0.001)),
# tf.keras.layers.Dense(d*bei, activation='relu'),
tf.keras.layers.Dropout(0.3),
tf.keras.layers.Dense(1, activation='sigmoid')])        
        
# Create an optimizer with the desired parameters.
myopt = tf.keras.optimizers.SGD(learning_rate = SGD_rate)

model.compile(optimizer=myopt, loss='binary_crossentropy', metrics=['accuracy'])
        
model.fit(train_x, train_ground_truth_class,  epochs=epochs, batch_size=batchsize, verbose = 0)


# Evaluate the model on the test data using `evaluate`
#print("Evaluate on test data")
#results = model.evaluate(val_x, val_ground_truth_class)
#print("test loss, test acc:", results)

prediction2 = model.predict(val_x)


Then, print(prediction2) are all NAN, why?
